container-images/tcib/base/os/rally/rally.yaml

```diff
@@ -0,0 +1,21 @@
+tcib_envs:
+  USE_EXTERNAL_FILES: true
+tcib_actions:
+- run: bash /usr/local/bin/uid_gid_manage {{ tcib_user }}
+- run: dnf -y install {{ tcib_packages['common'] | join(' ') }} && dnf clean all && rm -rf /var/cache/dnf
+- run: cp /usr/share/tcib/container-images/tcib/base/os/rally/rally_sudoers /etc/sudoers.d/rally_sudoers
+- run: chmod 440 /etc/sudoers.d/rally_sudoers
+- run: mkdir -p /var/lib/rally/external_files
+- run: mkdir -p /var/lib/kolla/config_files
+- run: chown -R rally.rally /var/lib/rally
+- run: touch /var/lib/kolla/config_files/config.json
+- run: cp /usr/share/tcib/container-images/tcib/base/os/rally/run_rally.sh /var/lib/rally/run_rally.sh
+- run: chmod +x /var/lib/rally/run_rally.sh
+
+tcib_entrypoint: /var/lib/rally/run_rally.sh
+
+tcib_packages:
+  common:
+  - openstack-rally
```

Comment:
don't we need to include openstack-rally-plugins[1] rpm?

[1] https://github.com/rdo-packages/rally-openstack-distgit/blob/rpm-master/openstack-rally-plugins.spec#L15

Reply:
i honestly have no idea, I've never used rally .. my only goal is to have that image ready once someone needs it and if someone does need it, they can build the rest of the script (or the list of the dependencies) according their needs

Reply:
Ok so no one requested it already? afaik that's used together considering previous OSP releases so asked about that. The original request that resulted into this image should have the required information.

Reply:
@eduolivares can you confirm this so it can get included here.

Reply:
There are two sets of downstream jobs using rally now.

The neutron jobs running rally use the rally infrared plugin. That plugin allows to install rally using three different methods: git, pip and rpm.
https://github.com/redhat-openstack/infrared/tree/master/plugins/rally/tasks/setup
The default method is pip:
https://github.com/redhat-openstack/infrared/blob/master/plugins/rally/plugin.spec#L36
If I'm not wrong, installing extra plugins is always done with git:
https://github.com/redhat-openstack/infrared/blob/master/plugins/rally/tasks/setup/rpm/main.yml#L35
And, actually, the neutron job does not add any of these extra plugins. This is from a downstream job:
```
2024-06-20 23:17:22.673 | TASK [Clone Rally-Plugins git repo] ********************************************
2024-06-20 23:17:22.674 | task path: /home/rhos-ci/jenkins/workspace/DFG-network-neutron-17.1_director-rhel-virthost-3cont_2comp-ipv6-all-geneve-dvr-stf-rally/infrared/plugins/rally/tasks/setup/pip/main.yml:34
2024-06-20 23:17:22.676 | Thursday 20 June 2024  23:17:21 +0000 (0:00:29.075)       0:01:00.118 ********* 
2024-06-20 23:17:22.679 | skipping: [undercloud-0] => {
2024-06-20 23:17:22.681 |     "changed": false,
2024-06-20 23:17:22.683 |     "skip_reason": "Conditional result was False"
2024-06-20 23:17:22.684 | }
```

Regarding the other downstream jobs using rally, the perfscale jobs, it seems they install browbeat from git at this moment. If it can be installed from RPMs, that would be much better, I guess.


So, sorry for the long comment, but, in conclussion, if plugins (at least browbeat) can be installed from RPMs, that would be nice.

Reply:
@eduolivares thx for the response what i meant was specific to rally-openstack inclusion in the image and looking infrared, as defaults is pip, i see downstream rally jobs using that only and  have rally-openstack installed always as per https://github.com/redhat-openstack/infrared/blob/master/plugins/rally/tasks/setup/pip/main.yml#L17-L32, so would have been good to include the rpm for it in the image as well, but since it's already merged can be a follow up when needed :)